src/app/task/shared/models/task.model.ts

```diff
@@ -1,11 +1,12 @@
 export class Task {
+  id: string | undefined;
```

Comment:
id should not be undefined

Reply:
Is the problem specifically that it can have no value? or can it be null? Because when calling new Task ir create-task component, i want it to be passed to BE with no id, so that mongodb can automatically asign it a unique id.

Reply:
Yes, you're right, ID should never be undefined, of course when you create a new task it won't have an ID (but it will be generated in mongodb, you can use Angular's Omit to omit the id field from the Task object, because technically whenever you fetch task details the Task object will never have ID = undefined